components/cli/ballerina/bal_executor.go

```diff
@@ -0,0 +1,117 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http:www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package ballerina
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/cellery-io/sdk/components/cli/pkg/constants"
+	"github.com/cellery-io/sdk/components/cli/pkg/util"
+)
+
+type BalExecutor interface {
+	Build(fileName string, iName []byte) error
+	Version() (string, error)
+	ExecutablePath() (string, error)
+}
+
+type LocalBalExecutor struct {
+}
+
+// NewLocalBalExecutor returns a LocalBalExecutor instance.
+func NewLocalBalExecutor() *LocalBalExecutor {
+	balExecutor := &LocalBalExecutor{}
+	return balExecutor
+}
+
+// Build executes ballerina build on an executable bal file.
+func (baleExecutor *LocalBalExecutor) Build(fileName string, iName []byte) error {
+	exePath, err := baleExecutor.ExecutablePath()
+	if err != nil {
+		return fmt.Errorf("failed to get executable path, %v", err)
+	}
+	cmd := exec.Command(exePath, "run", fileName, "build", string(iName), "{}")
+	err = cmd.Start()
+	if err != nil {
+		return fmt.Errorf("error occurred while starting to build image, %v", err)
+	}
+	err = cmd.Wait()
+	if err != nil {
+		return fmt.Errorf("error occurred while waiting to build image, %v", err)
+	}
+	return nil
+}
+
+// Version returns the ballerina version.
+func (baleExecutor *LocalBalExecutor) Version() (string, error) {
+	version := ""
+	cmd := exec.Command("ballerina", "version")
+	stdoutReader, _ := cmd.StdoutPipe()
+	stdoutScanner := bufio.NewScanner(stdoutReader)
+	go func() {
+		for stdoutScanner.Scan() {
+			version += stdoutScanner.Text()
+		}
+	}()
+	err := cmd.Start()
+	if err != nil {
+		return "", fmt.Errorf("error starting to get ballerina version, %v", err)
+	}
+	err = cmd.Wait()
+	if err != nil {
+		return "", fmt.Errorf("error waiting to get ballerina version, %v", err)
+	}
+	return version, nil
+}
+
+// ExecutablePath returns the ballerina executable path.
+func (baleExecutor *LocalBalExecutor) ExecutablePath() (string, error) {
+	var err error
+	var ballerinaVersion string
+	if ballerinaVersion, err = baleExecutor.Version(); err != nil {
+		return "", fmt.Errorf("failed to check ballerina version, %v", err)
+	}
+	if strings.Contains(ballerinaVersion, "Ballerina") {
+		if len(strings.Split(ballerinaVersion, " ")) > 0 {
+			if strings.Split(ballerinaVersion, " ")[1] == constants.BALLERINA_VERSION {
+				// If existing ballerina version is as the expected version, execute ballerina run without executable path
+				return "ballerina", nil
```

Comment:
ballerina is hardcoded in multiple places, move to a constant.

Reply:
Updated.